packages/react/src/createReactQueryHooks.tsx

```diff
@@ -44,28 +45,87 @@ interface TRPCUseQueryBaseOptions extends TRPCRequestOptions {
   ssr?: boolean;
 }
 
-interface UseTRPCQueryOptions<TError, TOutput, TQueryKey extends unknown[]>
-  extends UseQueryOptions<TOutput, TError, TOutput, TQueryKey>,
+interface TRPCUseQueryBaseOptionsOptionalInput<TInput> {
+  input?: TInput;
+}
+
+interface TRPCUseQueryBaseOptionsRequiredInput<TInput> {
+  input: TInput;
+}
+
+interface UseTRPCQueryOptions<TPath, TInput, TOutput, TError>
+  extends UseQueryOptions<TOutput, TError, TOutput, [TPath, TInput]>,
     TRPCUseQueryBaseOptions {}
 
-interface UseTRPCInfiniteQueryOptions<
-  TError,
-  TOutput,
-  TQueryKey extends unknown[],
-> extends UseInfiniteQueryOptions<TOutput, TError, TOutput, TOutput, TQueryKey>,
+interface UseTRPCQueryOptionalInputOptions<TPath, TInput, TOutput, TError>
+  extends UseQueryOptions<TOutput, TError, TOutput, [TPath, TInput]>,
+    TRPCUseQueryBaseOptionsOptionalInput<TInput> {}
+
+interface UseTRPCQueryRequiredInputOptions<TPath, TInput, TOutput, TError>
+  extends UseQueryOptions<TOutput, TError, TOutput, [TPath, TInput]>,
+    TRPCUseQueryBaseOptions,
+    TRPCUseQueryBaseOptionsRequiredInput<TInput> {}
+
+interface UseTRPCInfiniteQueryOptions<TPath, TInput, TOutput, TError>
+  extends UseInfiniteQueryOptions<
+      TOutput,
+      TError,
+      TOutput,
+      TOutput,
+      [TPath, TInput]
+    >,
     TRPCUseQueryBaseOptions {}
 
+interface UseTRPCInfiniteQueryRequiredInputOptions<
+  TPath,
+  TInput,
+  TOutput,
+  TError,
+> extends UseInfiniteQueryOptions<TOutput, TError, TOutput, [TPath, TInput]>,
+    TRPCUseQueryBaseOptions,
+    TRPCUseQueryBaseOptionsRequiredInput<TInput> {}
+
+// interface UseTRPCInfiniteQueryOptionalInputOptions<
+//   TPath,
+//   TInput,
+//   TOutput,
+//   TError,
+// > extends UseInfiniteQueryOptions<TOutput, TError, TOutput, [TPath, TInput]>,
+//     TRPCUseQueryBaseOptions,
+//     TRPCUseQueryBaseOptionsOptionalInput<TInput> {}
+
```

Comment:
Will be added at a later point - a bit trickier as infinite queries always are at least `{ cursor: any }`